presto-orc/src/main/java/com/facebook/presto/orc/ScanUtils.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.orc;
+
+import java.util.Arrays;
+
+public class ScanUtils
+{
+    private static final int MIN_INCREMENT = 100;
+
+    private ScanUtils() {}
+
+    /**
+     *Creates an int array with size rounded up. Intended for use
+     * when the maximum size is known for the current use. Some extra
+     * is reserved for the case where the next use is larger. Do not
+     ** use this when expecting the size to grow dynamically.
+     */
+    public static int[] newIntArrayForReuse(int size)
```

Comment:
Looking at the usages, I'm wondering if new int[getInitialSize(numRows)], new byte[getInitialSize(numRows)], etc. would be easier to read. This way, we'll need only one helper function for allocating new arrays.

```
... = new int[getInitialSize(numRows)];

    public static int getInitialSize(int capacity)
    {
        return capacity + capacity / 4;
    }
```